hare/hare.go

```diff
@@ -108,6 +108,7 @@ type Hare struct {
 	broker     *Broker
 	sign       *signing.EdSigner
 	blockGenCh chan LayerOutput
+	syncState  system.SyncStateProvider // provider function to check if the node is currently synced
```

Comment:
it's less error prone to delegate the synced'ness check to broker to ensure they check for the the same condition.

Reply:
I've extracted what previously constituted the broker's  `Synced` to a single method of the syncState, so I think the current approach is pretty much equivalent to what you are suggesting now.